### ISSUE TITLE
Cannot close selection dropdown after press outside.

### ISSUE THREAD

**Issue by pqminh** [toystars/react-native-multiple-select#49]
## Issue summary
Cannot close selection dropdown after press outside.

### Library versions
List the version of react-native and multi-select in use as below
"react-native": "0.47.2",
"react-native-multiple-select": "^0.3.0",

### Help wanted
I think the dropdown list should be close when press outside. First, open the dropdown list, it will onFocus into search input. And when onBlur search Input so the dropdown list will close.

**Comment by msafayasar** [comment]
Did you find a solution?

**Comment by violabg** [comment]
We definitely need a way to close the dropdown even if you don't select an option

**Comment by lucifurtun** [comment]
Is there any plan to fix this issue ?

**Comment by toystars** [comment]
Please use the latest version. There ought to be a dropdown icon that can be pressed to close the dropdown

**Comment by sokratisvs** [comment]
@toystars is it documented cause i cannot find any info for this.

**Comment by lucifurtun** [comment]
@toystars, you mean `0.4.4` ? I don't see one. Btw, I'm talking about `single` mode.

**Comment by toystars** [comment]
@lucifurtun yes, the latest version is `0.4.4`. The c`close` icon should also work in single mode

**Comment by kristkirp** [comment]
@toystars, I don't see either the close icon, what's the prop i can use to close?

**Comment by nicocrm** [comment]
With `hideSubmitButton` it allows closing the dropdown by tapping the arrow, but I am still looking for a way to automatically close it when tapping outside of the selection box

**Comment by PravinLinge** [comment]
Is any body got solution?

**Comment by ashraf-ainia** [comment]
I just found this workaround using the component callbacks using the React functional component:
first, you have to wrap your main screen with the TouchableWithoutFeedback component.

```
const multiSelectRef = useRef();
const closeMultiSelectIfOpened = () => {
    if (multiSelectRef.current.state.selector) {
      multiSelectRef.current._toggleSelector();
    }
  }

return(
<TouchableWithoutFeedback onPress={closeMultiSelectIfOpened}>
  <View style={styles.screen}>
        <View style={{ flex: 1 }}>
            <MultiSelect ref={multiSelectRef} ........ />
         </View>
          ...............
   </View>
</TouchableWithoutFeedback>
);
```

**Comment by tadinhlam1307** [comment]
> I just found this workaround using the component callbacks using the React functional component:
> first, you have to wrap your main screen with the TouchableWithoutFeedback component.
> 
> ```
> const multiSelectRef = useRef();
> const closeMultiSelectIfOpened = () => {
>     if (multiSelectRef.current.state.selector) {
>       multiSelectRef.current._toggleSelector();
>     }
>   }
> 
> return(
> <TouchableWithoutFeedback onPress={closeMultiSelectIfOpened}>
>   <View style={styles.screen}>
>         <View style={{ flex: 1 }}>
>             <MultiSelect ref={multiSelectRef} ........ />
>          </View>
>           ...............
>    </View>
> </TouchableWithoutFeedback>
> );
> ```

i change to:   <MultiSelect onToggleList={() => closeMultiSelectIfOpened()} ........ /> 
it work with me

**Comment by mehranali12** [comment]
no work

**Comment by xgokhankose** [comment]
> I just found this workaround using the component callbacks using the React functional component: first, you have to wrap your main screen with the TouchableWithoutFeedback component.
> 
> ```
> const multiSelectRef = useRef();
> const closeMultiSelectIfOpened = () => {
>     if (multiSelectRef.current.state.selector) {
>       multiSelectRef.current._toggleSelector();
>     }
>   }
> 
> return(
> <TouchableWithoutFeedback onPress={closeMultiSelectIfOpened}>
>   <View style={styles.screen}>
>         <View style={{ flex: 1 }}>
>             <MultiSelect ref={multiSelectRef} ........ />
>          </View>
>           ...............
>    </View>
> </TouchableWithoutFeedback>
> );
> ```
its working but flatlist scroll working so slowly , anyone know ?
